src/orquestra/sdk/schema/ir.py

```diff
@@ -201,6 +217,10 @@ class TaskDef(BaseModel):
     # An empty list [] means a Task with no parameters
     parameters: t.Optional[t.List[TaskParameter]]
 
+    # Statically inferred from the task function. See also `TaskOutputMetadata`'s
+    # docstring.
+    output_metadata: TaskOutputMetadata
```

Comment:
**thought**: this is a backwards incompatible change. It's likely that users won't be able to interact with old workflows after upgrading the SDK/CLI version.

However, any bugfix for this PR's motivating issue is likely to change how we process workflow graphs, so it would be backwards-incompatible anyway.